File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/db/hero/HeroDatabase.java
package com.akexorcist.repositoryarchcomponents.db.hero;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.akexorcist.repositoryarchcomponents.api.hero.response.Hero;
import com.akexorcist.repositoryarchcomponents.api.hero.response.VoteResult;

/**
 * Created by Akexorcist on 10/6/2017 AD.
 */

@Database(entities = {Hero.class, VoteResult.class}, version = 1)
public abstract class HeroDatabase extends RoomDatabase {
    abstract public HeroDao heroDao();

    abstract public VoteDao voteDao();
}

File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/db/hero/HeroDatabase.kt
package com.akexorcist.repositoryarchcomponents.hero

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import com.akexorcist.repositoryarchcomponents.api.hero.respond.Hero
import com.akexorcist.repositoryarchcomponents.api.hero.respond.VoteResult

/**
 * Created by Anirut Teerabut on 12/23/2017.
 */
@Database(entities = arrayOf(Hero::class, VoteResult::class), version = 1)
abstract class HeroDatabase : RoomDatabase() {

    abstract fun heroDao(): HeroDao

    abstract fun voteDao(): VoteDao
}